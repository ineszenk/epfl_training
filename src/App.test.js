import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import ReuseableForm from "./App";

const handleSubmit = jest.fn();

const theForm = (
  <ReuseableForm onSubmit={handleSubmit}>
    {handleChange => (
      <React.Fragment>
        <fieldset>
          <legend>Street address</legend>
          <input name="address" type="text" onChange={handleChange} />
        </fieldset>
        <fieldset>
          <legend>Postal code</legend>
          <input name="postal-code" type="number" onChange={handleChange} />
        </fieldset>
      </React.Fragment>
    )}
  </ReuseableForm>
);

const wrapper = mount(theForm);

describe("the <ReuseableForm /> component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(theForm).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it("renders the children", () => {
  expect(wrapper.contains(theForm)).toBe(true);
});
it("handles a change", () => {
  const inputName = "address";
  const inputValue = "EPFL Innovation Park, Building E";
  wrapper.find(`input[name='${inputName}']`).simulate("change", {
    target: { name: inputName, value: inputValue }
  });
  expect(wrapper.state(inputName)).toBe(inputValue);
});
