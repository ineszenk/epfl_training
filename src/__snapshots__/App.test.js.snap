// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <ReuseableForm /> component renders correctly 1`] = `null`;
